File: CourseWork/LexicalAnalysis/TokenType.cs
namespace CourseWork.LexicalAnalysis
{
    public enum TokenType
    {
        DbDirective,
        DwDirective,
        DdDirective,
        EquDirective,
        IfDirective,
        ElseDirective,
        EndifDirective,

        Instruction,
        SegmentKeyword,
        EndKeyword,
        EndsKeyword,

        Register32,
        Register16,
        Register8,

        Symbol,

        Identifier,

        HexNumber,
        DecNumber,
        BinNumber,
        Text,

        UserSegment,
        SegmentRegister,
        Label,
    }
}

File: java/src/main/java/course/work/sp/tokenizer/TokenType.java
package course.work.sp.tokenizer;

public enum TokenType {
    DbDir,
    DwDir,
    DdDir,
    DbIdentifier,
    DwIdentifier,
    DdIdentifier,
    SegmentIdentifier,

    Instruction,
    JmpWord,
    JncWord,
    SegmentWord,
    EndWord,
    EndsWord,

    Reg8,
    Reg16,
    Reg32,

    Symbol,
    Comma,

    Identifier,

    HexNumber,
    DecNumber,
    BinNumber,

    Text,

    BytePtr,
    DwordPtr,
    WordPtr,
    Ptr,

    SegmentRegister,
    KeyWord,
    Label,

    Mem8,
    Mem16,
    Mem32,
    Mem,

    Imm,

    Unknown,
    Empty,

    DS,
    CS,
    SS,
    ES,
    GS,
    FS

}
